src/server/lib/okta/openid-connect.ts

```diff
@@ -0,0 +1,242 @@
+/* eslint-disable functional/immutable-data */
+import { Issuer, IssuerMetadata, Client } from 'openid-client';
+import { randomBytes } from 'crypto';
+import { Request } from 'express';
+import { joinUrl } from '@guardian/libs';
+import { PersistableQueryParams } from '@/shared/model/QueryParams';
+import { getConfiguration } from '@/server/lib/getConfiguration';
+import { getProfileUrl } from '@/server/lib/getProfileUrl';
+import { ResponseWithRequestState } from '@/server/models/Express';
+import { logger } from '@/server/lib/serverSideLogger';
+import { RoutePaths } from '@/shared/model/Routes';
+
+/**
+ * An object to hold the state the user was in while performing the
+ * Authorization Code flow.
+ * - `nonce` - Used as the `state` parameter in the Authorization Code Flow
```

Comment:
Is it worth linking out to the docs that explain the `state` param ? (https://developer.okta.com/docs/reference/api/oidc/#parameter-details)